enclave-runtime/src/tests.rs

```diff
@@ -127,648 +131,313 @@ pub extern "C" fn test_main_entrance() -> size_t {
 	)
 }
 
-#[allow(unused)]
-pub fn ensure_no_empty_shard_directory_exists() {
-	// ensure no empty states are within directory (created with init-shard)
-	// otherwise an 'index out of bounds: the len is x but the index is x'
-	// error will be thrown
-	let shards = state::list_shards().unwrap();
-	for shard in shards {
-		if !state::exists(&shard) {
-			state::init_shard(&shard).unwrap();
-		}
-	}
-}
-
-#[allow(unused)]
 fn test_compose_block_and_confirmation() {
 	// given
-	ensure_no_empty_shard_directory_exists();
-	let latest_onchain_header =
-		Header::new(1, Default::default(), Default::default(), [69; 32].into(), Default::default());
-	let call_hash: H256 = [94; 32].into();
-	let call_hash_two: H256 = [1; 32].into();
-	let signed_top_hashes = [call_hash, call_hash_two].to_vec();
-	let shard = ShardIdentifier::default();
-	let state_hash_apriori: H256 = [199; 32].into();
-	// ensure state starts empty
-	state::init_shard(&shard).unwrap();
-	let mut state = Stf::init_state();
+	let (_, mut state, shard, _) = test_setup();
+
+	let signed_top_hashes: Vec<H256> = vec![[94; 32].into(), [1; 32].into()].to_vec();
+
 	Stf::update_sidechain_block_number(&mut state, 3);
 
 	// when
 	let (opaque_call, signed_block) = crate::compose_block_and_confirmation::<Block, SignedBlock>(
-		&latest_onchain_header,
+		&latest_parentchain_header(),
 		signed_top_hashes,
 		shard,
-		state_hash_apriori,
+		state.hash(),
 		&mut state,
 	)
 	.unwrap();
-	let xt_block_encoded = [TEEREX_MODULE, BLOCK_CONFIRMED].encode();
-	let block_hash_encoded = blake2_256(&signed_block.block().encode()).encode();
-	let mut opaque_call_vec = opaque_call.encode();
 
 	// then
+	let expected_call = OpaqueCall::from_tuple(&(
+		[TEEREX_MODULE, BLOCK_CONFIRMED],
+		shard,
+		blake2_256(&signed_block.block().encode()),
+	));
+
 	assert!(signed_block.verify_signature());
 	assert_eq!(signed_block.block().block_number(), 4);
-	assert!(opaque_call_vec.starts_with(&xt_block_encoded));
-	let mut stripped_opaque_call = opaque_call_vec.split_off(xt_block_encoded.len());
-	assert!(stripped_opaque_call.starts_with(&shard.encode()));
-	let stripped_opaque_call = stripped_opaque_call.split_off(shard.encode().len());
-	assert!(stripped_opaque_call.starts_with(&block_hash_encoded));
+	assert!(opaque_call.encode().starts_with(&expected_call.encode()));
 
 	// clean up
 	state::tests::remove_shard_dir(&shard);
 }
 
-#[allow(unused)]
 fn test_submit_trusted_call_to_top_pool() {
 	// given
```

Comment:
Thanks for letting these live, I appreciate that 🍡

Reply:
All live is precious!